xUnique.py

```diff
@@ -217,21 +229,12 @@ def sort_pbxproj(self, sort_pbx_by_file_name=False):
         pbx_flag = False
         last_two = deque([])
 
-        def file_dir_cmp(x, y):
-            if '.' in x:
-                if '.' in y:
-                    return cmp(x, y)
-                else:
-                    return 1
-            else:
-                if '.' in y:
-                    return -1
-                else:
-                    return cmp(x, y)
+        def file_dir_order(x):
+            return ('.' in x, x)
```

Comment:
You are wrong, the next code snippet
`python -c "file_dir_order = (lambda x: ('.' in x, x)); print('\n'.join(name for name in sorted(['dir1', 'last_dir', 'file.m', 'a.h'], key=file_dir_order)))"`
will print:
`dir1
last_dir
a.h
file.m`
(all dirs before files)